### PR TITLE
Update Dockerfile to make retraining easier.

### PR DESCRIPTION
**Pull request by tushuhei**
- Generate `scavenger_classes.ts` for the retrained model. (solves #32 )
- Update README to instruct replacing `scavenger_classes.ts`.
- Solve the issue that TensorFlow GPU is overwritten by other
  submodules.
- Move bottolneck dir under /data to make rerun easy.

### CODE REVIEW

File: training/Dockerfile.gpu [hunk 1]
```diff
@@ -11,20 +11,28 @@ RUN apt-get update && \
...
 
-RUN pip3 --no-cache-dir install \
-        tensorflow-gpu \
-        tensorflowjs
-
 WORKDIR /
 
 RUN git clone --depth 1 https://github.com/tensorflow/tensorflow.git --branch r1.7
 
+# Installing tensorflowjs package after training since it may overwrite the GPU version of TensorFlow.
```
Comment (jbruwer) [inline training/Dockerfile.gpu]: What line is this comment referring to? Seems out of place here when the full training follows.

File: README.md [hunk 2]
```diff
@@ -74,8 +74,10 @@ They are SavedModel files in a web-friendly format converted by the
 You can build your own game using your own custom image recognition model by replacing
 the corresponding files under the `dist/model/` directory with the newly generated ones.
 
-You also need to update `src/js/scavenger_classes.ts` in order to update the
-label outputs from the custom model with human-readable strings.
+The training script also generates `scavenger_classes.ts` for your custom model.
+Replace `src/js/scavenger_classes.ts` with it and build the app to run the game
```
Comment (jbruwer) [inline README.md]: Nit: Maybe we can make this language a little more explicit. How about the following.

"The training script will also generate a file called `scavenger_classes.ts` which works in conjunction with your generated custom model. You need to replace the file at `src/js/scavenger_classes.ts` with this newly generated `scavenger_classes.ts` file so that the labels of your model match with the trained data. After replacing the file you can run the build script normally to test your model in a browser. See the `README` file for information on running a preview server"

Comment (tushuhei) [reply README.md]: Thank you for your suggestion! done.

File: training/Dockerfile.gpu (hunk 1 above)
Comment (tushuhei) [reply training/Dockerfile.gpu]: I moved this comment to right before the `pip install` line for clearer reference.

**Comment by tushuhei** [comment]
Updated! Please take another look.

File: training/Dockerfile.gpu [hunk 3]
```diff
@@ -11,20 +11,27 @@ RUN apt-get update && \
...
-        tensorflowjs
-
 WORKDIR /
 
 RUN git clone --depth 1 https://github.com/tensorflow/tensorflow.git --branch r1.7
 
 CMD python3 /tensorflow/tensorflow/examples/image_retraining/retrain.py \
         --image_dir /data/images \
-        --how_many_training_steps=10000 \
+        --how_many_training_steps=4000 \
```
Comment (jbruwer) [inline training/Dockerfile.gpu:20]: Just curious what the affect is of training less than before. Does the model still converge at this training step? Assuming you did a end to end test on these new docker files to make sure the output is correct?

Comment (tushuhei) [reply training/Dockerfile.gpu:20]: I run another training with this configuration on our training dataset and confirmed that the default number of training steps (4000) provides good convergence. I could remove this line since it's the default value, but I kept it because a good parameter value really depends on the number of training images and objects to classify.

**Review by jbruwer — APPROVED** [review]
